src/app/utils/date.ts

```diff
@@ -31,3 +31,27 @@ export const createQueryWithDateParam = (
   // Start and end date is missing, return empty object.
   return {}
 }
+
+/**
+ * Returns start of day. If date is not given, the start
+ * of day will be calculated based on current date.
+ *
+ * @param option to contain optional date and timezone used to calculate the
+ * starting day. Defaults to today's date and Singapore timezone
+ */
+export const getStartOfDay = (option?: {
+  date?: Date
+  timezone?: string
+}): Date => {
+  if (!option) {
+    option = {}
+  }
+  if (!option.date) {
+    option.date = new Date()
+  }
+  if (!option.timezone) {
+    option.timezone = 'Asia/Singapore'
+  }
```

Comment:
may I propose using the magic of destructuring:
```suggestion
  option = {
    date: new Date()
    timezone: 'Asia/Singapore'
    ...option
  }
```

Reply:
Wow, TIL, thanks for this!